### PR TITLE
Avalon related fixes

### PR DESCRIPTION
**Pull request by Dananji**
This PR contains;
- fixes to parsing canvas URIs which were assumed to follow a specific format => now Ramp works with more generic canvas URIs
- changes to identification of transcripts based on a combination of file extension and content-type of the fetch request
- changes to progress-bar component to enable auto-advance for multi-canvas manifests
- fixes player disposing and re-initializing process
- cleanup of the testing and sample data in the repository

This work was done at the initial stage of bringing Ramp into avalon.

Closes avalonmediasystem/avalon#5169

### CODE REVIEW

**Review by cjcolvar — COMMENTED** [review]
Looks good.  I really only have one question about canvas ids.

File: src/components/ListItem.js [hunk 1]
```diff
@@ -37,13 +39,21 @@ const ListItem = ({ item, isTitle }) => {
...
   };
 
   const isClickable = () => {
-    const itemId = getItemId(item);
     const timeFragment = getMediaFragment(itemId, playerRange.end);
-    const isCanvas = canvasIndex + 1 == getCanvasId(itemId);
+    let isCanvas = false;
+    if (canvasIndex != undefined) {
+      const currentCanvasId = canvasesInManifest(manifest)[canvasIndex];
+      isCanvas = currentCanvasId.split('/').reverse()[0] == getCanvasId(itemId);
```
[the remark this answers is not in the document]
Comment (Dananji) [reply src/components/ListItem.js:55]: Yes, this still assumes a format. I agree using the full URL as id could be more accurate and appropriate for more use-cases. I'll change that. Thanks for pointing that out :100: This method sounds way more reliable than what I've been trying to do with these canvas IDs.

File: src/components/MediaPlayer/VideoJSPlayer.js [hunk 2]
```diff
@@ -380,14 +385,23 @@ function VideoJSPlayer({
...
-      const segmentId = getItemId(segment);
-      const segmentCanvas = getCanvasId(segmentId) - 1;
-      const segmentRange = getMediaFragment(segmentId, canvasDuration);
-      const isInRange = checkSrcRange(segmentRange, playerRange);
-      const isInSegment =
-        currentTime >= segmentRange.start && currentTime < segmentRange.end;
-      if (isInSegment && isInRange && segmentCanvas == canvasIndex) {
-        return segment;
+      const { id, isTitleTimespan } = segment;
+      const canvasId = id.split('#t')[0];
```
[the remark this answers is not in the document]
Comment (Dananji) [reply src/components/MediaPlayer/VideoJSPlayer.js:389]: Good catch! it seems like a refactoring mishap by mistake. I will fix it.

File: src/components/ListItem.js [hunk 3]
```diff
@@ -37,13 +39,21 @@ const ListItem = ({ item, isTitle }) => {
 
     playerDispatch({ clickedUrl: e.target.href, type: 'navClick' });
 
-    manifestDispatch({ item: item, type: 'switchItem' });
+    let navItem = {
+      id: itemId,
+      label: getLabelValue(item.label),
+      isTitletimespan: isChild || isTitle
```
[the remark this answers is not in the document]
Comment (Dananji) [reply src/components/ListItem.js:45]: Good catch!

**Pull request merged**